src/state_machine/workload.zig

```diff
@@ -503,6 +522,78 @@ pub fn WorkloadType(comptime AccountingStateMachine: type) type {
         }
 
+        fn build_get_account_filter(self: *const Self, body: []tb.AccountFilter) usize {
+            assert(body.len == 1);
+            const account_filter = &body[0];
+            account_filter.* = std.mem.zeroInit(tb.AccountFilter, .{});
+
+            account_filter.account_id = if (self.auditor.pick_account(.{
+                .created = true,
+                .debits_must_not_exceed_credits = null,
+                .credits_must_not_exceed_debits = null,
+                .has_transfers = true,
+            })) |account| account.id else
+            // Pick an account with valid index (rather than "random.int(u128)") because the
+            // Auditor must decode the id to check for a matching account.
+            self.auditor.account_index_to_id(self.random.int(usize));
+
+            switch (self.random.enumValue(enum { none, debits, credits, all })) {
+                .none => {}, // Testing invalid flags.
+                .debits => account_filter.flags.debits = true,
+                .credits => account_filter.flags.credits = true,
+                .all => {
+                    account_filter.flags.debits = true;
+                    account_filter.flags.credits = true;
+                },
+            }
+            account_filter.flags.reversed = self.random.boolean();
+
+            const batch_size = AccountingStateMachine.constants.batch_max.get_account_transfers;
+            const transfer_count: u32 = if (self.auditor.get_account_state(
+                account_filter.account_id,
+            )) |account_state| transfer_count: {
+                const dr_transfer_count: u32 = if (account_filter.flags.debits)
+                    account_state.dr_transfer_count
+                else
+                    0;
+                const cr_transfer_count: u32 = if (account_filter.flags.credits)
+                    account_state.cr_transfer_count
+                else
+                    0;
+                break :transfer_count dr_transfer_count + cr_transfer_count;
+            } else 0;
+
+            switch (self.random.enumValue(enum { none, one, all, max })) {
+                .none => account_filter.limit = 0, // Testing invalid limit.
+                .one => account_filter.limit = 1,
+                .all => account_filter.limit = transfer_count,
+                .max => account_filter.limit = batch_size,
+            }
+
+            // Apply filter by timestamp only when we the prefix or the suffix of the expected
+            // result can be asserted. The filter range is always restrictive.
+            if (account_filter.flags.debits and account_filter.flags.credits and
+                account_filter.limit > 1 and transfer_count < batch_size and
+                chance(self.random, self.options.account_filter_timestamp_range_probability))
+            {
+                if (self.auditor.get_account_state(
+                    account_filter.account_id,
+                )) |account_state| {
```

Comment:
In this branch, we _should_ have account state, right? THen maybe `.?` instead of an if here, to make it fail loudly in case of some future unrelated change?